sdk/identity/Azure.Identity/src/Credentials/DefaultAzureCredentialOptions.cs

```diff
@@ -67,6 +67,11 @@ public class DefaultAzureCredentialOptions : TokenCredentialOptions
         /// </summary>
         public ResourceIdentifier ManagedIdentityResourceId { get; set; }
 
+        /// <summary>
+        /// Specifies Powershell process timeout in milliseconds.
+        /// </summary>
+        public int PowerShellProcessTimeoutMs { get; set; } = (int)TimeSpan.FromSeconds(10).TotalMilliseconds;
```

Comment:
I'm wondering if we should generalize this to work for all process based credentials (Visual Studio, and AzureCli). 

@schaabs any thoughts ?

Reply:
This sounds reasonable, you mean calling this something like "SubProcessTimeOutMs" and then sharing this timeout between all providers using subproceses ? this sounds simple enough, but also wondering if clients would ever have the need to control each provider separately with its own timeout. 

I'm inclined to do the simplest approach of sharing single SubProcessTimeout since will add value soon and not complicate a lot the API so that users can configure this. Then evaluate later if we need to have greater control over each provider timeout.

thoughts ?

Reply:
@christothes @schaabs ping

Reply:
Sorry for the delay on this issue. I am thinking we make it `TimeSpan DeveloperCredentialTimeout` which refers to the "Developer Credentials in [this flow](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity#defaultazurecredential). I'd also adjust the comments to name the specific credentials that are affected (all the orange ones, currently). 

cc @scottaddie